### PR TITLE
Add support for specifying CNI networks in podman play kube

### PR DESCRIPTION
**Pull request by cfelder**
As far as I understood it is not possible to define container network on the yaml level (https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.18/#pod-v1-core).

I am adding networking options to ``podman play cube`` though.

### CODE REVIEW

**Comment by mheon** [comment]
I don't know if we want to do this - should we be sourcing from the Kube YAML instead? @haircommander thoughts?

**Comment by cfelder** [comment]
@mheon I would prefer specifying this on yaml level but I have not found an option to declare a network for a pod in the specs.

**Comment by haircommander** [comment]
yeah I wouldn't go this route. There are some first class pod API features that relate to pod networking:

the[dnsConfig](https://kubernetes.io/docs/concepts/services-networking/dns-pod-service/#pod-s-dns-config) object would take care of `--dns` `--dns-opt` and `--dns-search`
[hostAliases](https://kubernetes.io/docs/concepts/services-networking/add-entries-to-pod-etc-hosts-with-host-aliases/#adding-additional-entries-with-hostaliases) seem to take care of `--add-hosts` and `--ip`

finally, you can replicate `--publish` on the container level with the [ports](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.18/#containerport-v1-core) object

there doesn't seem to be a way to specify `--mac-address`, and `--network` also seems inaccessible. It is also hard to figure out how similar behavior to `--no-hosts` could be implemented.

**Comment by haircommander** [comment]
I suppose `--network` could be added as a hacky way to specify a network (Because k8s api doesn't provide it). But the others should be done the k8s way IMO

**Comment by cfelder** [comment]
I am more than happy to implement this in yaml if you can help me with the specification as I don't know how to specify the network options in yaml syntax.

it would be a lot easier to implement this in ``PlayKubeYAML`` in ``pkg/adapter/pods.go`` using ``libPod.WithPodNetworks``.

**Comment by haircommander** [comment]
> I am more than happy to implement this in yaml if you can help me with the specification as I don't know how to specify the network options in yaml syntax.
> 

were the above links not enough to get started? do you have specific questions about them?

> it would be a lot easier to implement this in `PlayKubeYAML` in `pkg/adapter/pods.go` using `libPod.WithPodNetworks`.

As an example, when iterating through the container objects, you can access `container.Ports` and take the information defined [here](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.18/#containerport-v1-core) and translate it into the equivalent values for libpod/options:WithNetNS()

does that help?

**Comment by haircommander** [comment]
> > I am more than happy to implement this in yaml if you can help me with the specification as I don't know how to specify the network options in yaml syntax.
> 
> were the above links not enough to get started? do you have specific questions about them?
> 
> > it would be a lot easier to implement this in `PlayKubeYAML` in `pkg/adapter/pods.go` using `libPod.WithPodNetworks`.
> 
> As an example, when iterating through the container objects, you can access `container.Ports` and take the information defined [here](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.18/#containerport-v1-core) and translate it into the equivalent values for libpod/options:WithNetNS()
> 
> does that help?

nvm on that, we already do something similar in `getPodPorts` in pkg/adapter/pods.go . but hopefully that serves as an example for the other options

**Comment by cfelder** [comment]
Thanks for your remarks I changed my PR to just add ``--network`` for ``podman play kube`` in order to support specifying CNI networks which cannot be done in yaml syntax right now.

**Comment by mheon** [comment]
Concur that just --network is appropriate...

We probably also want to restrict it to only listing a set of CNI networks - we should not support `--net=host` and `--net=bridge` as those are handled in the YAML.

File: pkg/adapter/pods.go [hunk 1]
```diff
@@ -343,7 +343,7 @@ func (r *LocalRuntime) CreatePod(ctx context.Context, cli *cliconfig.PodCreateVa
 			logrus.Debugf("Pod will use host networking")
 			options = append(options, libpod.WithPodHostNetwork())
 		case "":
-			return "", errors.Errorf("invalid value passed to --net: must provide a comma-separated list of CNI networks or host")
+			return "", errors.Errorf("invalid value passed to --network: must provide a comma-separated list of CNI networks or host")
```
Comment (TomSweeneyRedHat) [inline pkg/adapter/pods.go]: host is not mentioned  in the man page.  It should be removed here or added there.

File: docs/source/markdown/podman-play-kube.1.md [hunk 2]
```diff
@@ -36,6 +36,10 @@ The [username[:password]] to use to authenticate with the registry if required.
 If one or both values are not supplied, a command line prompt will appear and the
 value can be entered.  The password is entered without echo.
 
+**--network**=*cni networks*
+
+A comma-separated list of the names of CNI networks the pod should join.
+
```
Comment (TomSweeneyRedHat) [inline docs/source/markdown/podman-play-kube.1.md]: an example of valid input might be nice in the example section.

**Comment by TomSweeneyRedHat** [comment]
Any tests possible?  Code LGTM, just a few nitty doc things.  Thanks @cfelder !

File: pkg/adapter/pods.go (hunk 1 above)
Comment (cfelder) [reply pkg/adapter/pods.go]: This belongs to ``CreatePod`` which includes host in its man page but I've corrected this in line 601.

I just wanted to correct ``--net`` to ``--network`` here.

File: docs/source/markdown/podman-play-kube.1.md (hunk 2 above)
Comment (cfelder) [reply docs/source/markdown/podman-play-kube.1.md]: Done

**Comment by vrothberg** [comment]
Ping. Let's get this in.

**Comment by mheon** [comment]
/approve
LGTM

**Pull request merged**